code.test.js

```diff
@@ -0,0 +1,77 @@
+// test.
+const fs = require('fs');
+eval(fs.readFileSync('code.js')+'');
+
+
+// Test cases
+const testCases = [
+    {
+        graph1: [
+            [0, 1, 0],
+            [1, 0, 1],
+            [0, 1, 0],
+        ],
+        graph2: [
+            [0, 1, 1],
+            [1, 0, 0],
+            [1, 0, 0],
+        ],
+        expected: false, // Not isomorphic
+    },
+    {
+        graph1: [
+            [0, 1, 1],
+            [1, 0, 1],
+            [1, 1, 0],
+        ],
+        graph2: [
+            [0, 1, 1],
+            [1, 0, 1],
+            [1, 1, 0],
+        ],
+        expected: true, // Isomorphic
+    },
+    {
+        graph1: [
+            [0, 1],
+            [1, 0],
+        ],
+        graph2: [
+            [0, 1],
+            [1, 0],
+        ],
+        expected: true, // Isomorphic
+    },
+    {
+        graph1: [
+            [0, 1, 0],
+            [1, 0, 0],
+            [0, 0, 0],
+        ],
+        graph2: [
+            [0, 1, 1],
+            [1, 0, 0],
+            [1, 0, 0],
+        ],
+        expected: false, // Not isomorphic
+    },
+    {
+        graph1: [
+            [0, 1, 0],
+            [1, 0, 1],
+            [0, 1, 0],
+        ],
+        graph2: [
+            [0, 1, 1],
+            [1, 0, 0],
+            [1, 0, 0],
+        ],
+        expected: false, // Not isomorphic
+    },
+];
+
+// Run test cases
+testCases.forEach(({ graph1, graph2, expected }, index) => {
+    const result = are_isomorphic(graph1, graph2);
+    console.log(`Test Case ${index + 1}:`, result === expected ? "Passed" : "Failed");
```

Comment:
This will not fail the check if a test fails -- could you change that please?

Reply:
Oh I'm sorry about that I will go through and change that. I'll modify things from that point if there are further issues. Thank you

Reply:
I see my error I realize that my code does not pass all of my checks I'm sorry I didn't catch that before, I will go back and work on my code some more to debug and finish up this assingment. Again sorry for that mistake, thank you for helping me by pointing that out.

Reply:
I realized that my error was in my checkIsomorphicMapping function. In this function instead of returning true for just one valid mapping I needed to return true if and only if all of the mappings were valid. I had origionally returned true if only one of them was true, which is not the case with isomorphism. The updated function now returns true only if all mappings are valid, and returns false when there is an invalid mapping. I also fixed my test code from where it did not stop running if it was a failed test. I believe everything should be up to date!

Reply:
Your test code still doesn't look right. It should throw an error when a test fails (you can use the `assert` function for this for example). At the moment, you're going to get a green check mark here even if all tests fail.

Reply:
Oh sorry I will get that fixed!

Reply:
I believe I fixed it, I added in the assert at the end of my test code